File: cardbattlemodule/src/main/java/com/syahputrareno975/cardbattlemodule/interfaces/LobbyStreamEvent.java
package com.syahputrareno975.cardbattlemodule.interfaces;

import com.syahputrareno975.cardbattlemodule.model.player.PlayerModel;
import com.syahputrareno975.cardbattlemodule.model.playerWithCard.PlayerWithCardsModel;

import java.util.ArrayList;

public interface LobbyStreamEvent {
    void onConnected(PlayerModel p,LobbyStreamEventController c);
    void onPlayerJoin(PlayerModel p);
    void onGetOnePlayer(PlayerModel p);
    void onGetAllPlayer(ArrayList<PlayerModel> p );
    void onGetPlayerData( PlayerWithCardsModel p);
    void onError(String e);
    void onPlayerLeft(PlayerModel p);
    void onPlayerSuccessLeft();
    void onDisconnected();
    void onException(String e,int flag ,LobbyStreamEventController c);
}

File: app/src/main/java/com/syahputrareno975/simplecardbattle/interfaces/LobbyStreamEvent.kt
package com.syahputrareno975.simplecardbattle.interfaces

import com.syahputrareno975.simplecardbattle.model.player.PlayerModel
import com.syahputrareno975.simplecardbattle.model.playerWithCard.PlayerWithCardsModel

interface LobbyStreamEvent {
    fun onConnected(p : PlayerModel,c : LobbyStreamEventController)
    fun onPlayerJoin(p : PlayerModel)
    fun onGetOnePlayer(p : PlayerModel)
    fun onGetAllPlayer(p : ArrayList<PlayerModel>)
    fun onGetPlayerData(p : PlayerWithCardsModel)
    fun onError(e : String)
    fun onPlayerLeft(p : PlayerModel)
    fun onPlayerSuccessLeft()
    fun onDisconnected()
    fun onException(e : String,flag : Int,c : LobbyStreamEventController)
}